### PR TITLE
update conv layers in cugraph-dgl for pylibcugraphops 23.04

### PR DESCRIPTION
**Pull request by tingyu66**
Update cugraph-ops models to use pylibcugraphops 23.04. This PR also supersedes #3264 

CC: @MatthiasKohl

### CODE REVIEW

File: ci/test_python.sh [hunk 1]
```diff
@@ -117,7 +117,7 @@ if [[ "${RAPIDS_CUDA_VERSION}" == "11.8.0" ]]; then
       --channel "${PYTHON_CHANNEL}" \
       --channel pytorch \
       --channel pytorch-nightly \
-      --channel dglteam/label/cu117 \
+      --channel dglteam/label/cu118 \
```
Comment (VibhuJawa) [inline ci/test_python.sh:120]: Ahh, just saw that that they have cu118 builds here: https://anaconda.org/dglteam/dgl 


Surprisingly it is missing on  getting started page: https://www.dgl.ai/pages/start.html

**Review by MatthiasKohl — COMMENTED** [review]
looks good from my side. as discussed live, with 23.06, hopefully, we can get rid of the `pad_offsets` thing altogether, using new structures

**Review by ajschmidt8 — APPROVED** [review]
Approving `ops-codeowner` file changes

**Review by VibhuJawa — APPROVED** [review]
LGTM

**Review by rlratzel — APPROVED** [review]
LGTM

**Pull request merged**